applications/StructuralMechanicsApplication/custom_constitutive/small_strain_isotropic_plasticity_factory_3d.h

```diff
@@ -0,0 +1,280 @@
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+    * Default constructor.
+    */
+    SmallStrainIsotropicPlasticityFactory3D()
+    {
+    }
+
+    /**
+    * Clone.
+    */
+    ConstitutiveLaw::Pointer Clone() const override
+    {
+        return Kratos::make_shared<SmallStrainIsotropicPlasticityFactory3D>(*this);
+    }
+
+    /**
+    * Copy constructor.
+    */
+    SmallStrainIsotropicPlasticityFactory3D (const SmallStrainIsotropicPlasticityFactory3D& rOther)
+    : ConstitutiveLaw(rOther)
+    {
+    }
+    /**
+    * Destructor.
+    */
+    ~SmallStrainIsotropicPlasticityFactory3D() override
+    {
+    }
+
+    /**
+     * creates a new constitutive law pointer
+     * @param NewParameters The configuration parameters of the new constitutive law
+     * @return a Pointer to the new constitutive law
+     */
+    ConstitutiveLaw::Pointer Create(Kratos::Parameters NewParameters) const override
+    {
+        const std::string& yield = NewParameters["yield_surface"].GetString();
+        const std::string& potential = NewParameters["plastic_potential"].GetString();
+
+        KRATOS_ERROR_IF(yield == "SimoJu")  << "SimoJu yield surface not available in plasticity "  << yield << std::endl;
+        KRATOS_ERROR_IF(yield == "Rankine") << "Rankine yield surface not available in plasticity " << yield << std::endl;
+
+        const std::string& name = "SmallStrainIsotropicPlasticity3D" + yield + potential;
+        return KratosComponents<ConstitutiveLaw>::Get(name).Clone();
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * Computes the material response in terms of 1st Piola-Kirchhoff stresses and constitutive tensor
+     * @see Parameters
+     */
+    void CalculateMaterialResponsePK1(ConstitutiveLaw::Parameters& rValues) override
+    {
+        this->CalculateMaterialResponseCauchy(rValues);
+    }
+
+    /**
+     * Computes the material response in terms of 2nd Piola-Kirchhoff stresses and constitutive tensor
+     * @see Parameters
+     */
+    void CalculateMaterialResponsePK2(ConstitutiveLaw::Parameters& rValues) override
+    {
+        this->CalculateMaterialResponseCauchy(rValues);
+    }
+
+    /**
+     * Computes the material response in terms of Kirchhoff stresses and constitutive tensor
+     * @see Parameters
+     */
+    void CalculateMaterialResponseKirchhoff(ConstitutiveLaw::Parameters& rValues) override
+    {
+        this->CalculateMaterialResponseCauchy(rValues);
+    }
+
+    /**
+     * Computes the material response in terms of Cauchy stresses and constitutive tensor
+     * @see Parameters
+     */
+    void CalculateMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues) override
+    {
+        KRATOS_ERROR << "The base factory class has been called, define a correct constitutive law " << std::endl;
+    } 
+
+    /**
+     * to be called at the end of each solution step
```

Comment:
I would not add the function descriptions to overridden functions
=> as soon as the description in the baseclass changes this will be outdated